packages/api/src/users/commands/handlers/updateUser.handler.ts

```diff
@@ -0,0 +1,23 @@
+import { BadRequestException } from '@nestjs/common';
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+
+import { UserErrorMessage } from '@coderscamp/shared/api';
+import type { RegisteredUser } from '@coderscamp/shared/models';
+
+import { UsersRepository } from '../../users.repository';
+import { UpdateUserCommand } from '../updateUser.command';
+
+@CommandHandler(UpdateUserCommand)
+export class UpdateUserHandler implements ICommandHandler<UpdateUserCommand> {
+  constructor(private repository: UsersRepository) {}
+
+  async execute({ input }: UpdateUserCommand): Promise<RegisteredUser> {
```

Comment:
In which use cases we update user? In such generic command we lose context.
For example: 
when user change email address you should send email in order to verify ownership of the email address.
It's a lot easier with more granular - domain-centric events like: UserEmailAddressWasChanged instead of comapring state to get known what has changed.